src/libraries/System.Linq/src/System/Linq/Count.cs

```diff
@@ -59,19 +60,45 @@ public static int Count<TSource>(this IEnumerable<TSource> source, Func<TSource,
                 ThrowHelper.ThrowArgumentNullException(ExceptionArgument.predicate);
             }
 
-            int count = 0;
-            foreach (TSource element in source)
+            return source.TryGetSpan(out ReadOnlySpan<TSource> span)
+                ? CountSpan(span, predicate)
+                : CountEnumerable(source, predicate);
+
+            [StackTraceHidden]
+            static int CountSpan(ReadOnlySpan<TSource> source, Func<TSource, bool> predicate)
             {
-                checked
+                int count = 0;
+                foreach (TSource element in source)
                 {
-                    if (predicate(element))
+                    checked
```

Comment:
You do not need `checked` here. The Span can only have so many elements.

Reply:
Thanks.